services/intersection-api/api/src/main/java/us/dot/its/jpo/ode/api/controllers/AssessmentController.java

```diff
@@ -0,0 +1,263 @@
+import us.dot.its.jpo.ode.api.accessors.assessments.LaneDirectionOfTravelAssessment.LaneDirectionOfTravelAssessmentRepository;
+import us.dot.its.jpo.ode.api.accessors.assessments.SignalStateAssessment.StopLineStopAssessmentRepository;
+import us.dot.its.jpo.ode.api.accessors.assessments.SignalStateEventAssessment.SignalStateEventAssessmentRepository;
+import us.dot.its.jpo.ode.mockdata.MockAssessmentGenerator;
+
+@Slf4j
+@RestController
+@ConditionalOnProperty(
+    name = "enable.api",
+    havingValue = "true",
+    matchIfMissing = false
+)
+@ApiResponses(
+    value = {
+        @ApiResponse(responseCode = "200", description = "Successful operation"),
+        @ApiResponse(responseCode = "401", description = "Unauthorized"),
+        @ApiResponse(responseCode = "403", description = "Forbidden"),
+        @ApiResponse(responseCode = "404", description = "Not Found"),
+        @ApiResponse(responseCode = "500", description = "Internal Server Error")
+    }
+)
+public class AssessmentController {
+
+    @Autowired
+    LaneDirectionOfTravelAssessmentRepository laneDirectionOfTravelAssessmentRepo;
+
+    @Autowired
+    ConnectionOfTravelAssessmentRepository connectionOfTravelAssessmentRepo;
+
+    @Autowired
+    StopLineStopAssessmentRepository stopLineStopAssessmentRepo;
+
+    @Autowired
+    SignalStateEventAssessmentRepository signalStateEventAssessmentRepo;
+
+
+    @Operation(summary = "Get Connection of Travel Assessments", description = "Get Connection of Travel Assessments, filtered by intersection ID, start time, and end time. The latest flag will only return the latest message satisfying the query.")
+    @RequestMapping(value = "/assessments/connection_of_travel", method = RequestMethod.GET, produces = "application/json")
+    @PreAuthorize("@PermissionService.isSuperUser() || (@PermissionService.hasIntersection(#intersectionID) and @PermissionService.hasRole('USER'))")
+    public ResponseEntity<List<ConnectionOfTravelAssessment>> findConnectionOfTravelAssessment(
+            @RequestParam(name = "intersection_id", required = false) Integer intersectionID,
+            @RequestParam(name = "start_time_utc_millis", required = false) Long startTime,
+            @RequestParam(name = "end_time_utc_millis", required = false) Long endTime,
+            @RequestParam(name = "latest", required = false, defaultValue = "false") boolean latest,
+            @RequestParam(name = "test", required = false, defaultValue = "false") boolean testData) {
+
+        if (testData) {
+            List<ConnectionOfTravelAssessment> list = new ArrayList<>();
+            list.add(MockAssessmentGenerator.getConnectionOfTravelAssessment());
+            return ResponseEntity.ok(list);
+        } else {
+            Query query = connectionOfTravelAssessmentRepo.getQuery(intersectionID, startTime, endTime, latest);
+            long count = connectionOfTravelAssessmentRepo.getQueryResultCount(query);
```

Comment:
issue(blocking): this count call executes the query a 2nd time just to log the size of the returned dataset in debug mode.

Reply:
I resolved additional patterns of this in other places as well